build.mill

```diff
@@ -219,7 +219,32 @@ object Deps {
   }
 }
 
-def millVersion: T[String] = T { VcsVersion.vcsState().format() }
+def millVersion: T[String] = T {
+  val vcsVersion = VcsVersion.vcsState().format()
+  val isCI = System.getenv("CI") != null
+  if (!isCI && (vcsVersion.contains("DIRTY") || vcsVersion.count(_ == '-') >= 2)) {
+    // vcsVersion looks like "0.11.11-6-49d084-DIRTYa0693949", we only keep
+    // the "0.11.11" part here
+    val baseVersion = vcsVersion.takeWhile(_ != '-')
+    val lastDotIndex = baseVersion.lastIndexOf('.')
+    if (lastDotIndex < 0)
+      // unexpected shape, keeping vcsVersion
+      vcsVersion
+    else {
+      val patchVersionOpt = baseVersion.substring(lastDotIndex + 1).toIntOption
+      patchVersionOpt match {
+        case None =>
+          // unexpected shape, keeping vcsVersion
+          vcsVersion
+        case Some(patchVersion) =>
+          val base = baseVersion.substring(0, lastDotIndex)
+          s"$base.${patchVersion + 1}-SNAPSHOT"
+      }
+    }
+  }
+  else
+    vcsVersion
+}
```

Comment:
I might misreading this longish blob, but I guess what you want is:

```suggestion
def millVersion: T[String] = T {
  VcsVersion.vcsState().format(dirtySep = "-SNAPSHOT", dirtyHashDigits = 0)
}
```

Reply:
For reference: https://github.com/lefou/mill-vcs-version?tab=readme-ov-file#formatting-options